### DISCUSSION TITLE
Compilation issues

### DISCUSSION THREAD

**Discussion by Forever-chardonnay** [jedisct1/libsodium#1349] [Q&A]
l met a problem, i cannot find the interface in wasm when using the zig compiled,which was described in the ci, it's seems like all the interface was not exported at compile.
Compilation method zig build -Dtarget=wasm32-wasi mentioned in the ci file.
Is there any compilation method that exposes all interfaces?

**Comment by jedisct1** [comment]
Hi,

What functions did you find missing?

**Reply by Forever-chardonnay** [reply]
After compiling the wasm file, configure an independent non-web runtime environment，then call wasm_runtime_lookup_function to find function,return fail.
